omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/LocationResource1_8.java

```diff
@@ -249,39 +251,54 @@ public void purge(Location location, RequestContext context) throws ResponseExce
 	protected NeedsPaging<Location> doGetAll(RequestContext context) {
 		return new NeedsPaging<Location>(Context.getLocationService().getAllLocations(context.getIncludeAll()), context);
 	}
-	
+
 	/**
 	 * @see org.openmrs.module.webservices.rest.web.resource.impl.DelegatingCrudResource#doSearch(org.openmrs.module.webservices.rest.web.RequestContext)
-	 *      A query string and/or a tag uuid can be passed in; if both are passed in, returns an
+	 *      A query string and/or a tag (referenced by name or uuid) can be passed in; if both are passed in, returns an
 	 *      intersection of the results; excludes retired locations
 	 */
 	@Override
 	protected PageableResult doSearch(RequestContext context) {
-		
-		LocationService locationService = Context.getLocationService();
-		
-		String tagUuid = context.getParameter("tag");
+
+		String tag = context.getParameter("tag");
 		String query = context.getParameter("q");
-		
-		List<Location> locationsByTag = null;
-		List<Location> locationsByQuery = null;
-		
-		if (tagUuid != null) {
-			LocationTag locationTag = locationService.getLocationTagByUuid(tagUuid);
-			locationsByTag = locationService.getLocationsHavingAllTags(Arrays.asList(locationTag));
+
+		List<Location> locationsByTag = Collections.emptyList();
+		List<Location> locationsByQuery = Collections.emptyList();
```

Comment:
I would set these to null here by default.  That will resolve the issue you highlight below...